src/Telemetry/AzureDevOpsTimeline/AzureDevOpsTimeline.cs

```diff
@@ -248,27 +248,34 @@ private async Task RunLoop(CancellationToken cancellationToken)
                     new KustoValue("BuildId", b.Id.ToString(), KustoDataTypes.Int),
                     new KustoValue("Status", b.Status, KustoDataTypes.String),
                     new KustoValue("Result", b.Result, KustoDataTypes.String),
-                    new KustoValue("Repository", b.Repository.Name, KustoDataTypes.String),
+                    new KustoValue("Repository", b.Repository?.Name ?? b.Repository?.Id, KustoDataTypes.String),
                     new KustoValue("Reason", b.Reason, KustoDataTypes.String),
                     new KustoValue("BuildNumber", b.BuildNumber, KustoDataTypes.String),
                     new KustoValue("QueueTime", b.QueueTime, KustoDataTypes.DateTime),
                     new KustoValue("StartTime", b.StartTime, KustoDataTypes.DateTime),
                     new KustoValue("FinishTime", b.FinishTime, KustoDataTypes.DateTime),
-                    new KustoValue("Project", b.Project.Name, KustoDataTypes.String),
+                    new KustoValue("Project", b.Project?.Name, KustoDataTypes.String),
+                    new KustoValue("DefinitionId", b.Definition?.Id.ToString(), KustoDataTypes.String),
+                    new KustoValue("Definition", $"{b.Definition?.Path}\\{b.Definition?.Name}", KustoDataTypes.String),
                 });
 
             _logger.LogInformation("Saving TimelineValidationMessages...");
             await KustoHelpers.WriteDataToKustoInMemoryAsync(
                 ingest,
                 options.KustoDatabase,
-                "TimelineValidationMessages",
+                "TimelineIssues",
                 _logger,
                 validationResults,
                 b => new[]
                 {
                     new KustoValue("BuildId", b.buildId.ToString(), KustoDataTypes.Int),
-                    new KustoValue("Result", b.validationResult.Result, KustoDataTypes.String),
+                    new KustoValue("RecordId", null, KustoDataTypes.String),
+                    new KustoValue("Index", null, KustoDataTypes.Int),
+                    new KustoValue("Path", null, KustoDataTypes.String),
+                    new KustoValue("Type", b.validationResult.Result, KustoDataTypes.String),
+                    new KustoValue("Category", "ValidationResult", KustoDataTypes.String),
                     new KustoValue("Message", b.validationResult.Message, KustoDataTypes.String),
+                    new KustoValue("Bucket", null, KustoDataTypes.String),
```

Comment:
I feel like we're never actually going to look at Category, Bucket is the thing we actually will look at and give people a way to modify.  Maybe we should make the bucket "ValidationResult" too?  Also, we might want to make the Bucket "Code" if Category is "Code" and bucket is empty?

Reply:
Oh true, I can set the bucket to validation too.  Though it's tricky, because it's 2 problems in 1 (but there isn't anything we can do about it, I suppose). It's either "I couldn't talk to GitHub" or "the yaml is bad".  Maybe we should... encourage... Azure DevOps to provide some sort of differentiation between those two things.

Reply:
Fair.  At this point, though, I don't think we need to get that granular.  In the future, maybe we will.  That is an interesting case though because "I couldn't talk to GitHub" is an external infrastructure issue and "the yaml is bad" is an internal developer issue.  Regardless, developers being bad at writing yaml is a known issue and probably isn't the big bucket thing we want to gather data on to resolve.